process/heartbeat/interceptedPeerAuthentication.go

```diff
@@ -0,0 +1,177 @@
+)
+
+// ArgInterceptedPeerAuthentication is the argument used in the intercepted peer authentication constructor
+type ArgInterceptedPeerAuthentication struct {
+	DataBuff             []byte
+	Marshalizer          marshal.Marshalizer
+	PeerSignatureHandler crypto.PeerSignatureHandler
+	Hasher               hashing.Hasher
+}
+
+type interceptedPeerAuthentication struct {
+	peerAuthentication   heartbeat.PeerAuthentication
+	peerId               core.PeerID
+	hash                 []byte
+	mutComputedShardID   sync.RWMutex
+	computedShardID      uint32
+	peerSignatureHandler crypto.PeerSignatureHandler
+}
+
+// NewInterceptedPeerAuthentication tries to create a new intercepted peer authentication instance
+func NewInterceptedPeerAuthentication(arg ArgInterceptedPeerAuthentication) (*interceptedPeerAuthentication, error) {
+	if len(arg.DataBuff) == 0 {
+		return nil, process.ErrNilBuffer
+	}
+	if check.IfNil(arg.Marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(arg.PeerSignatureHandler) {
+		return nil, process.ErrNilPeerSignatureHandler
+	}
+	if check.IfNil(arg.Hasher) {
+		return nil, process.ErrNilHasher
+	}
+
+	peerAuthentication, err := createPeerAuthentication(arg.Marshalizer, arg.DataBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	intercepted := &interceptedPeerAuthentication{
+		peerAuthentication:   *peerAuthentication,
+		peerSignatureHandler: arg.PeerSignatureHandler,
+	}
+
+	intercepted.processFields(arg.Hasher, arg.DataBuff)
+
+	return intercepted, nil
+}
+
+func createPeerAuthentication(marshalizer marshal.Marshalizer, buff []byte) (*heartbeat.PeerAuthentication, error) {
+	peerAuthentication := &heartbeat.PeerAuthentication{}
+	err := marshalizer.Unmarshal(peerAuthentication, buff)
+	if err != nil {
+		return nil, err
+	}
+
+	return peerAuthentication, nil
+}
+
+func (ipa *interceptedPeerAuthentication) processFields(hasher hashing.Hasher, buff []byte) {
+	ipa.hash = hasher.Compute(string(buff))
+	ipa.peerId = core.PeerID(ipa.peerAuthentication.Pid)
+}
+
+// PublicKey returns the public key as byte slice
+func (ipa *interceptedPeerAuthentication) PublicKey() []byte {
+	return ipa.peerAuthentication.Pubkey
+}
+
+// PeerID returns the peer ID
+func (ipa *interceptedPeerAuthentication) PeerID() core.PeerID {
+	return core.PeerID(ipa.peerAuthentication.Pid)
+}
+
+// Signature returns the signature done on the Pid field that can be checked against the provided PubKey field
+func (ipa *interceptedPeerAuthentication) Signature() []byte {
+	return ipa.peerAuthentication.Signature
+}
+
+// SetComputedShardID sets the computed shard ID. Concurrency safe.
+func (ipa *interceptedPeerAuthentication) SetComputedShardID(shardId uint32) {
+	ipa.mutComputedShardID.Lock()
+	ipa.computedShardID = shardId
+	ipa.mutComputedShardID.Unlock()
+}
+
+// CheckValidity will check the validity of the received peer heartbeat
+// However, this will not perform the signature validation since that can be an attack vector
```

Comment:
no need for such details. simply mention that it won't do sig checks

Reply:
done